dandiapi/api/views/dandiset.py

```diff
@@ -94,8 +95,21 @@ def create(self, request):
         if created:
             version_metadata.save()
 
-        dandiset = Dandiset()
-        dandiset.save()
+        if 'identifier' in serializer.validated_data['metadata']:
+            identifier = serializer.validated_data['metadata']['identifier']
+            if identifier.startswith('DANDI:'):
+                identifier = identifier[6:]
+            dandiset = Dandiset(id=int(identifier))
```

Comment:
We should probably add a try-except here (since something like `DANDI:qwerty` would throw an exception here), and if an exception is thrown, return a Response that informs them that the passed identifier is invalid.